src/decision/location_selection.jl

```diff
@@ -0,0 +1,151 @@
+- `area_to_seed` : area of coral to be seeded at each time step in km^2
+- `target_seed_sites` : list of candidate locations for seeding (indices)
+- `target_shade_sites` : list of candidate location to shade (indices)
+
+# Returns
+- `ranks_store` : number of scenarios * sites * 3 (last dimension indicates: site_id, seed rank, shade rank)
+    containing ranks for each scenario run.
+"""
+function run_site_selection(dom::Domain, scenarios::DataFrame, sum_cover::AbstractArray, area_to_seed::Float64; aggregate_function=nothing,
+    target_seed_sites=nothing, target_shade_sites=nothing)
+    ranks_store = NamedDimsArray(
+        zeros(nrow(scenarios), length(dom.site_ids), 3),
+        scenarios=1:nrow(scenarios),
+        sites=dom.site_ids,
+        ranks=["site_id", "seed_rank", "shade_rank"],
+    )
+
+    dhw_scens = dom.dhw_scens
+    wave_scens = dom.wave_scens
+
+    # Pre-calculate maximum depth to consider
+    scenarios[:, "max_depth"] .= scenarios.depth_min .+ scenarios.depth_offset
+    target_dhw_scens = unique(scenarios[:, "dhw_scenario"])
+    target_wave_scens = unique(scenarios[:, "wave_scenario"])
+
+    target_site_ids = Int64[]
+    if !isnothing(target_seed_sites)
+        append!(target_site_ids, target_seed_sites)
+    end
+
+    if !isnothing(target_shade_sites)
+        append!(target_site_ids, target_shade_sites)
+    end
+
+    n_sites = length(dom.site_ids)
+    for (scen_idx, scen) in enumerate(eachrow(scenarios))
+        depth_criteria = (dom.site_data.depth_med .<= scen.max_depth) .& (dom.site_data.depth_med .>= scen.depth_min)
+        depth_priority = findall(depth_criteria)
+
+        considered_sites = target_site_ids[findall(in(depth_priority), target_site_ids)]
+        ranks_store(scenarios=scen_idx, sites=dom.site_ids[considered_sites]) .= site_selection(
+            dom,
+            scen,
+            (mean(wave_scens[:, :, target_wave_scens], dims=(:timesteps, :scenarios)) .+ std(wave_scens[:, :, target_wave_scens], dims=(:timesteps, :scenarios))) .* 0.5,
+            (mean(dhw_scens[:, :, target_dhw_scens], dims=(:timesteps, :scenarios)) .+ std(dhw_scens[:, :, target_dhw_scens], dims=(:timesteps, :scenarios))) .* 0.5,
+            considered_sites,
+            sum_cover[scen_idx, :],
+            area_to_seed
+        )
+    end
+    if !isnothing(aggregate_function)
+        return ranks_store, aggregate_function(ranks_store, "seed"), aggregate_function(ranks_store, "shade")
+    else
+        return ranks_store
+    end
+end
+
+"""
+    ranks_to_location_order(ranks::NamedDimsArray, iv_type::String)
+
+
+Post-processing function for location ranks output of `run_location_selection()`. Gives the order 
+of location preference for each scenario as location ids.
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `iv_type` : String indicating the intervention type to perform aggregation on.
```

Comment:
Just a personal opinion here, but I think the name of this variable can be confusing. Maybe `interv_type` or `int_type` (or just `intervention_type`) would make the code easier to read (in my personal opinion). :)

Reply:
Hi @Zapiano, thanks for your comments, I'll incorporate them as I move this PR from draft. Apologies for some missing variables and documentation, I decided to split up a file and haven't cleaned up the structure yet.

 The PR is still a work in progress but I was hoping to get any requests/opinions on aggregate functions for the function `run_site_selection()` which are helpful for analysis or visualisation purposes. The current function outputs ranks, which gives where each site was ranked in terms of suitability for each intervention and scenario. This is not the most intuitive output to analyse, so I've added a rank frequency aggregation function (which gives the frequency with which each site was selected for a particular rank over all scenarios) and a rank order aggregation function (which lists the complete order of sites for each scenario). Let me know if you have other ideas for aggregation functions which could lead to interesting analysis of mcda outcomes :)

Reply:
@Rosejoycrocker oh, sorry, I haven't noticed it was a draft, my bad!

About the aggregation functions: thanks for the explanation! I was trying to understand the output of this function and what you just wrote helped a lot! I will think about it and let you know if I have any suggestions - or questions! :)

Reply:
That's all good, the comments are good anyway because the PR aims to make the function more intuitive to use, so it's all helpful :) Any other ideas for improving the usability of the function are welcome, particularly any aggregation methods you think might be useful for other anlyses you're working on.

Reply:
On the variable name: I'd prefer `interv_type` or to a smaller degree `intervention_type` (too verbose in my opinion.

Definitely not `int_type` because in my mind that relates to integer types (uint8, int16, int64, etc)

Reply:
Oh just to say, I was okay with `iv_type` - `iv` is the shorthand the ReefMod team uses so it's an established convention.

Reply:
Oh right, sorry misread the above. I'll change back to iv_type because it could be preferrable when we move to a more generic structure where intervention type will probably be a more widely used variable.